### PR TITLE
chore: Lit 3.x Upgrade

### PR DESCRIPTION
**Pull request by brian-patrick-3**
Notes:

1. I had to make a bunch of changes to the build scripts and update several dependencies to get this working. 
2. The only notable change in how we write our code is now the `accessor` keyword is required for decorators. 
   1. This will raise some conflicts in any active working branches you have, since I had to add `accessor` keywords to every component file.

### CODE REVIEW

**Comment by maulik-thaker1** [comment]
> Notes:
> 
> 1. I had to make a bunch of changes to the build scripts and update several dependencies to get this working.
> 2. The only notable change in how we write our code is now the `accessor` keyword is required for decorators.
>    
>    1. This will raise some conflicts in any active working branches you have, since I had to add `accessor` keywords to every component file.

Are we also upgrading LIT version to all other repo(s) like foundation and charts ?

**Comment by brian-patrick-3** [comment]
> Are we also upgrading LIT version to all other repo(s) like foundation and charts ?

All except content are done, see the other PRs.

**Review by maulik-thaker1 — APPROVED** [review]
LGTM.

File: .storybook/main.js [hunk 1]
```diff
@@ -9,7 +9,6 @@ export default {
   addons: [
     '@storybook/addon-links',
     '@storybook/addon-essentials',
-    '@storybook/addon-designs',
```
[the remark this answers is not in the document]
Comment (brian-patrick-3) [reply .storybook/main.js:12]: It is included in the core now.

File: src/components/global/header/headerLink.ts [hunk 2]
```diff
@@ -25,60 +25,60 @@ export class HeaderLink extends LitElement {
...
 
   /** Label for sub-menu link search input, which is visible with > 5 sub-links. */
   @property({ type: String })
-  searchLabel = 'Search';
+  accessor searchLabel = 'Search';
 
   /** Text for mobile "Back" button. */
   @property({ type: String })
-  backText = 'Back';
+  accessor backText = 'Back';
```
[the remark this answers is not in the document]
Comment (brian-patrick-3) [reply src/components/global/header/headerLink.ts:62]: It is not a breaking change, just the finalized standard for native decorators.

[the remark this answers is not in the document]
Comment (maulik-thaker1) [reply src/components/global/header/headerLink.ts:62]: but after merge this PR we need to apply this change into rest of all components right ?

[the remark this answers is not in the document]
Comment (maulik-thaker1) [reply src/components/global/header/headerLink.ts:62]: ok

**Pull request merged**